bundler/lib/bundler/resolver.rb

```diff
@@ -46,6 +46,8 @@ def start(requirements)
       @gem_version_promoter.prerelease_specified = @prerelease_specified = {}
       requirements.each {|dep| @prerelease_specified[dep.name] ||= dep.prerelease? }
 
+      @forced_requirements = requirements.select(&:force_version?).map {|dep| [dep.name, dep] }.to_h
```

Comment:
I think, if I understand this new approach, you'd need to handle `required_ruby_version` and `required_rubygems_version` specially. Probably by changing them to be unconstrained? Otherwise even if all the requirements of the forced version are prioritized, they would still not be met in the case of `required_ruby_version` and `required_rubygems_version`. Right?

This makes me wonder whether @simi's suggestion of providing a way of "monkeypatching requirements" a gemspec would be better:

```ruby
 gem "minitest", "~> 5.14" do |s|
  s.required_ruby_version = ">= 2.5"
end
```

It would also make it possible to use the feature multiple times in a `Gemfile` which I believe it could currently cause issues with this approach if they have conflicting requirements?

Reply:
I don't like the monkey patch syntax because it requires bringing all the parents into the equation in the force_version case and I don't think adds enough extra information to make it worth it. Also, I'm a bit wracking my brain in how to implement.

Here is an implementation of two additional flags to `gem`, `:override_ruby_version` and `:override_rubygems_version: https://github.com/lukaso/rubygems/pull/7 which I can fold into this PR.

The usage example is:

```ruby
gem 'rutabaga', git: 'git@github.com:simplybusiness/rutabaga.git', branch: 'ruby-24-or-less', override_ruby_version: true, override_rubygems_version: true
``` 

There is no easy or clear place to patch into the loading of `gemspecs` that I could find, so I had to place to overrides in a number of strategic places.

Reply:
I'm not sure about the implementation either, I can see how it could get super tricky, but if  we decided to do this, I was expecting to ship a single feature to accomplish this. Either `force_version: <exact_version>` as in: "ignore what you have to ignore during resolution to make sure you give me this specific version", or a more finer grained of monkeypatching then gemspec constraints used for resolution.

I'll think about it more.